### ISSUE TITLE
[FEATURE REQUEST] Add plugins support

### ISSUE THREAD

**Issue by a2nt** [KeitIG/museeks#392]
I'd like to listen and download music from youtube, but it will be against youtube policy so I'd like to create a plugin. Which will let me do it.

Maybe we can integrate YouTube to Museeks as https://github.com/headsetapp/headset-electron
## Pros:
+ It allows to browse music by channels and find something new
## Cons:
+ Headsetap doesn't allow to play local music and I don't like the way it looks.
+ No plugins supported
+ To follow YouTube policy  it opens extra window with video. "Due to a restriction in YouTube's policy, we can't separate the audio from the video and the player must be visible at all times."

Harmony looks better and closer to Museeks (https://github.com/vincelwt/harmony)
## Pros:
+ Plugins supported
+ Allows to play local music and music from external services (Reddit collections, YouTube and etc)
+ Doesn't opens an extra window to play YouTube video. Video is nicely integrated into interface
+ Plugins supported
## Cons:
+ It's not opensource
+ It doesn't support music collections as Headsetapp do

# So basically I'd like to have plugin extensions support to add external services.
+ Museeks shall have playlist bar at the left which can be hidden
+ HTML5 video area could be created at playlist bar
+ Playlists button toggles playlist bar
+ Plugins API shall let extensions to show/hide areas and bars, create own areas and toolbars, upload/download URLs, directly access files and file info (artist, name of the song, album, etc), create and update playlists, update application library, run console commands, play/pause/next/prev 

# Which plugins can be done
+ tag editor
+ external services play
+ download/upload to external services
+ converting tags to UTF8
+ website parsing

**Comment by KeitIG** [comment]
you might be interested in this discussion https://github.com/KeitIG/museeks/issues/260, closing this and maybe you can add your thoughts there.
